### PR TITLE
feat: Simplify go version selection

### PR DESCRIPTION
**Pull request by purelind**

### CODE REVIEW

File: jenkins/pipelines/ci/br/br_ghpr_unit_and_integration_test.groovy [hunk 1]
```diff
@@ -193,33 +193,20 @@ string trimPrefix = {
...
-    "go1.20": "${JOB_NAME}-go1200-${BUILD_NUMBER}",
-    "go1.21": "${JOB_NAME}-go1210-${BUILD_NUMBER}",
-]
+POD_GO_IMAGE = "hub.pingcap.net/jenkins/centos7_golang-1.21:latest"
+POD_LABEL = "${JOB_NAME}-${BUILD_NUMBER}-go121"
 
 node("master") {
     deleteDir()
-    def goversion_lib_url = 'https://raw.githubusercontent.com/PingCAP-QE/ci/main/jenkins/pipelines/goversion-select-lib.groovy'
+    def goversion_lib_url = 'https://raw.githubusercontent.com/PingCAP-QE/ci/main/jenkins/pipelines/goversion-select-lib-v2.groovy'
```
Comment (wuhuizuo) [inline jenkins/pipelines/ci/br/br_ghpr_unit_and_integration_test.groovy:201]: I suggest to save it in `libraries/tipipeline/vars` folder.

File: jenkins/pipelines/ci/br/br_ghpr_unit_and_integration_test.groovy [hunk 2]
```diff
@@ -193,33 +193,20 @@ string trimPrefix = {
...
     sh "curl --retry 3 --retry-delay 5 --retry-connrefused --fail -o goversion-select-lib.groovy  ${goversion_lib_url}"
     def goversion_lib = load('goversion-select-lib.groovy')
     GO_VERSION = goversion_lib.selectGoVersion(ghprbTargetBranch)
-    POD_GO_IMAGE = GO_IMAGE_MAP[GO_VERSION]
+    POD_GO_IMAGE = goversion_lib.selectGoImage(ghprbTargetBranch)
+    POD_LABEL = goversion_lib.getPodLabel(ghprbTargetBranch, JOB_NAME, BUILD_NUMBER)
```
Comment (wuhuizuo) [inline jenkins/pipelines/ci/br/br_ghpr_unit_and_integration_test.groovy:206]: can we put it in top level? such as:
```groovy
final REFS = gitee.composeRefFromPushEventPayload((env.jsonBody == "") ? params.jsonBody : env.jsonBody )
```

File: jenkins/pipelines/ci/br/br_ghpr_unit_and_integration_test.groovy (hunk 1 above)
Comment (purelind) [reply jenkins/pipelines/ci/br/br_ghpr_unit_and_integration_test.groovy:201]: These pipelines are gradually being deprecated, and the new ones will use Lib.

File: jenkins/pipelines/ci/br/br_ghpr_unit_and_integration_test.groovy (hunk 2 above)
Comment (purelind) [reply jenkins/pipelines/ci/br/br_ghpr_unit_and_integration_test.groovy:206]: you mean use `final` ?

**Pull request merged**